dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataIntegrityController.java

```diff
@@ -84,6 +81,21 @@ public WebMessage runDataIntegrity(
         @RequestParam( required = false ) List<String> checks,
         @CurrentUser User currentUser )
     {
+
+        //Do not include slow checks by default unless the user specifically requests them
+        if ( checks.isEmpty() )
+        {
+            for ( DataIntegrityCheck e : dataIntegrityService
+                .getDataIntegrityChecks() )
+            {
+                if ( !e.isSlow() )
+                {
+                    String name = e.getName();
+                    checks.add( name );
+                }
+            }
+        }
+
```

Comment:
Sorry, gave you a bad pointer there. Remembered now that the logic is different. The controller and the job configuration just use the empty list to mean "all" (so far). The place where we should filter this is here https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/dataintegrity/DefaultDataIntegrityService.java#L939, filter the returned set by the `isSlow` (please use streams :D)

Reply:
I think its in the right place now.